msmart/scanner.py

```diff
@@ -0,0 +1,255 @@
+        if ip is not None:
+            return self.get(ip)
+        return await self.get_all()
+
+    async def get_all(self):
+        await self._broadcast_message(self.count)
+        tasks = set()
+        while True:
+            
+            task = await self._get_response()
+            if task:
+                tasks.add(task)
+            else:
+                break
+        if len(tasks) > 0:
+            await asyncio.wait(tasks)
+            return [task.result() for task in tasks]
+        else:
+            return []
+
+    def get(self, ip):
+        self._send_message(ip)
+        loop = asyncio.new_event_loop()
+        try:
+            return loop.run_until_complete(self._get_response(ip))
+        finally:
+            loop.close()
+
+    async def _get_response(self, ip=None):
+        try:
+            data, addr = self.socket.recvfrom(512)
+            if ip is not None and addr[0] != ip:
+                return None
+            else:
+                ip = addr[0]
+            _LOGGER.debug("Midea Local Data {} {}".format(ip, data.hex()))
+            device = await scandevice.load(ip, data)
+            return asyncio.create_task(device.support_test(self.account, self.password))
+        except socket.timeout:
+            _LOGGER.debug("Socket timeout")
+            return None
+
+    async def _broadcast_message(self, amount):
+        nets = await _get_networks()
+        for i in range(amount):
+            for net in nets:
+                try:
+                    self.socket.sendto(
+                        BROADCAST_MSG, (str(net.broadcast_address), 6445)
+                    )
+                    self.socket.sendto(
+                        BROADCAST_MSG, (str(net.broadcast_address), 20086)
+                    )
+                    _LOGGER.debug("Broadcast message sent: " + str(i))
+                except:
+                    _LOGGER.debug("Unable to send broadcast to: " + str(net.broadcast_address))
+
+    async def _send_message(self, address):
+        self.socket.sendto(
+            BROADCAST_MSG, (address, 6445)
+        )
+        self.socket.sendto(
+            BROADCAST_MSG, (address, 20086)
+        )
+        _LOGGER.debug("Message sent")
+
+def gettoken(udpid, account, password):
+    global Client
+    if Client is None:
+        Client = cloud(account, password)
+    if Client.session == {}:
+        Client.login()
+    return Client.gettoken(udpid)
+    
+def _get_socket():
+    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+    sock.setsockopt(socket.SOL_SOCKET, socket.SO_BROADCAST, 1)
+    sock.settimeout(5)
+    return sock
+
+async def _get_networks():
+    nets :list() = []
+    adapters = ifaddr.get_adapters()
+    for adapter in adapters:
+        for ip in adapter.ips:
+            localNet = None
+            if ip.is_IPv4:
+                localNet = ipaddress.IPv4Network(f"{ip.ip}/{ip.network_prefix}", strict=False)
+            elif ip.is_IPv6:
```

Comment:
no need to check ipv6, ipv6 does not support broadcast addresses, and midea  does not support ipv6